src/MSALWrapper/AuthFlow/Broker.cs

```diff
@@ -84,58 +84,47 @@ protected override async Task<(TokenResult, IList<Exception>)> GetTokenInnerAsyn
         {
             IAccount account = await this.pcaWrapper.TryToGetCachedAccountAsync(this.preferredDomain)
                  ?? PublicClientApplication.OperatingSystemAccount;
-            this.logger.LogDebug($"Using cached account '{account?.Username}'");
 
+            TokenResult tokenResult = null;
             try
             {
-                try
+                tokenResult = await CachedAuth.TryCachedAuthAsync(
+                    this.logger,
+                    this.silentAuthTimeout,
+                    this.scopes,
+                    account,
+                    this.pcaWrapper,
+                    this.errors);
+
+                if (tokenResult == null)
                 {
                     try
                     {
-                        var tokenResult = await TaskExecutor.CompleteWithin(
+                        tokenResult = await TaskExecutor.CompleteWithin(
                             this.logger,
-                            this.silentAuthTimeout,
-                            "Get Token Silent",
-                            (cancellationToken) => this.pcaWrapper.GetTokenSilentAsync(this.scopes, account, cancellationToken),
+                            this.interactiveAuthTimeout,
+                            "Interactive Auth",
+                            (cancellationToken) => this.pcaWrapper
+                            .WithPromptHint(this.promptHint)
+                            .GetTokenInteractiveAsync(this.scopes, account, cancellationToken),

```

Comment:
The indentation here doesn't make it clear that these methods are being called on the `pcaWrapper`. This might also be moved into a helper method to make this code block smaller and more readable.